application/utils/harvester/chunk_record_builder.py

```
@@ -0,0 +1,107 @@
+    ) -> tuple[int, int]:
+        if not 0 <= start_char_idx < end_char_idx <= len(text):
+            raise ValueError("Chunk character offsets are outside the source document")
+        start_line = text.count("\n", 0, start_char_idx) + 1
+        end_position = end_char_idx - 1
+        end_line = text.count("\n", 0, end_position) + 1
+        return start_line, end_line
+
+
+def chunk_document(
+    document: Document,
+    config: ChunkingConfig | None = None,
```

Ruff (0.16.2)

[error] Undefined name `ChunkingConfig`

(F821)